ci/prow/make_config.go

```diff
@@ -0,0 +1,1010 @@
+pod_namespace: test-pods
+log_level: info
+
+branch-protection:
+  orgs:
+    knative:
+      # Protect all branches in knative
+      # This means all prow jobs with "always_run" set are required
+      # to pass before tide can merge the PR.
+      # Currently this is manually enabled by the knative org admins,
+      # but it's stated here for documentation and reference purposes.
+      protect: true
+      # Admins can overrule checks
+      enforce_admins: false
+
+tide:
+  queries:
+  - repos:
+    - knative/build
+    - knative/build-pipeline
+    - knative/build-templates
+    - knative/observability
+    - knative/serving
+    - knative/eventing
+    - knative/eventing-sources
+    - knative/docs
+    - knative/test-infra
+    - knative/pkg
+    - knative/caching
+    labels:
+    - lgtm
+    - approved
+    missingLabels:
+    - do-not-merge/hold
+    - do-not-merge/work-in-progress
+  merge_method:
+    knative: squash
+    knative/build-pipeline: rebase
+  target_url: https://prow.knative.dev/tide
+
+presets:
+# test run service account
+- labels:
+    preset-test-account: "true"
+  env:
+  - name: GOOGLE_APPLICATION_CREDENTIALS
+    value: [[.TestAccount]]
+  - name: E2E_CLUSTER_REGION
+    value: us-west1
+  volumes:
+  - name: account
+    secret:
+      secretName: test-account
+  volumeMounts:
+  - name: account
+    mountPath: /etc/test-account
+    readOnly: true
+# nightly release service account
+- labels:
+    preset-nightly-account: "true"
+  env:
+  - name: GOOGLE_APPLICATION_CREDENTIALS
+    value: [[.NightlyAccount]]
+  - name: E2E_CLUSTER_REGION
+    value: us-west1
+  volumes:
+  - name: account
+    secret:
+      secretName: nightly-account
+  volumeMounts:
+  - name: account
+    mountPath: /etc/nightly-account
+    readOnly: true
+# versioned release service account
+- labels:
+    preset-release-account: "true"
+  env:
+  - name: GOOGLE_APPLICATION_CREDENTIALS
+    value: [[.ReleaseAccount]]
+  - name: E2E_CLUSTER_REGION
+    value: us-west1
+  volumes:
+  - name: account
+    secret:
+      secretName: release-account
+  - name: hub-token
+    secret:
+      secretName: hub-token
+  volumeMounts:
+  - name: account
+    mountPath: /etc/release-account
+    readOnly: true
+  - name: hub-token
+    mountPath: /etc/hub-token
+    readOnly: true
+# backups service account
+- labels:
+    preset-backup-account: "true"
+  env:
+  - name: GOOGLE_APPLICATION_CREDENTIALS
+    value: /etc/backup-account/service-account.json
+  volumes:
+  - name: account
+    secret:
+      secretName: backup-account
+  volumeMounts:
+  - name: account
+    mountPath: /etc/backup-account
+    readOnly: true
+# docker-in-docker presets
+- labels:
+    preset-dind-enabled: "true"
+  env:
+  - name: DOCKER_IN_DOCKER_ENABLED
+    value: "true"
+  volumes:
+  - name: docker-graph
+    emptyDir: {}
+  volumeMounts:
+  - name: docker-graph
+    mountPath: /docker-graph
+`
+
+	// presubmitJob is the template for presubmit jobs.
```

Comment:
just an idea, how bout storing each job template in separate files. For example, copy paste line 285 to line 310 into a file located at `ci/prow/job_templates/presubmit_job_template.yaml`

Reply:
I thought about that, but was concerned that keeping them in separate files would make it harder to visualize common/shared settings/patterns, so I decided to keep them in a single place, at least for the initial version. There's a TODO in the header about factoring about some custom jobs from regular jobs; once that's done and the templates are simpler, I think that it would be a good thing keeping the templates in separate files.

Reply:
that totally make sense to me, since this is an initial version. This is much better than our 2224 lines config file

Reply:
Yep. Now adding a CI flow is just a 2-line PR. :D